hikari/impl/rest_bot.py

```diff
@@ -0,0 +1,518 @@
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Standard implementations of a Interaction based REST-only bot."""
+from __future__ import annotations
+
+__all__: typing.Sequence[str] = ["RESTBot"]
+
+import asyncio
+import logging
+import sys
+import typing
+
+from hikari import config
+from hikari import errors
+from hikari import traits
+from hikari.api import interaction_server as interaction_server_
+from hikari.impl import entity_factory as entity_factory_impl
+from hikari.impl import interaction_server as interaction_server_impl
+from hikari.impl import rest as rest_impl
+from hikari.internal import ux
+
+if typing.TYPE_CHECKING:
+    import concurrent.futures
+    import socket as socket_
+    import ssl
+
+    from hikari import applications
+    from hikari.api import entity_factory as entity_factory_
+    from hikari.api import rest as rest_
+    from hikari.api import special_endpoints
+    from hikari.interactions import bases as interaction_bases
+
+
+_LOGGER: typing.Final[logging.Logger] = logging.getLogger("hikari.rest_bot")
+
+
+class RESTBot(traits.RESTBotAware, interaction_server_.InteractionServer):
+    """Basic implementation of an interaction based REST-only bot.
+
+    Parameters
+    ----------
+    token : builtins.str
+        The bot or application token to use.
+    token_type : typing.Union[hikari.applications.TokenType, builtins.str]
+        The type of token being passed, this may be "Bot" or "Bearer".
+
+    Other Parameters
+    ----------------
+    allow_color : builtins.bool
+        Defaulting to `builtins.True`, this will enable coloured console logs
+        on any platform that is a TTY.
+        Setting a `"CLICOLOR"` environment variable to any **non `0`** value
+        will override this setting.
+
+        Users should consider this an advice to the application on whether it is
+        safe to show colours if possible or not. Since some terminals can be
+        awkward or not support features in a standard way, the option to
+        explicitly disable this is provided. See `force_color` for an
+        alternative.
+    application : typing.Optional[hikari.snowflakes.SnowflakeishOr[hikari.guilds.PartialApplication]]
+        Object or ID of the application this bot instance should be associated
+        with. If left as `builtins.None` then the client will try to work this
+        value out based on `token`.
```

Comment:
Was just going through the pr looking for something. Found this un-used docstring